src/test/groovy/com/bisnode/opa/client/ManagingDocumentSpec.groovy

```diff
@@ -0,0 +1,79 @@
+package com.bisnode.opa.client
+
+import com.bisnode.opa.client.data.OpaDataApi
+import com.bisnode.opa.client.data.OpaDocument
+import com.bisnode.opa.client.rest.ContentType
+import com.github.tomakehurst.wiremock.WireMockServer
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static com.bisnode.opa.client.rest.ContentType.Values.APPLICATION_JSON
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo
+import static com.github.tomakehurst.wiremock.client.WireMock.put
+import static com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+
+class ManagingDocumentSpec extends Specification {
+
+    private static int PORT = 8181;
+
+    private OpaDataApi client
+    private String host = "http://localhost:$PORT"
+    private WireMockServer wireMockServer = new WireMockServer(PORT)
+
+    def DOCUMENT = '{"example": {"flag": true}}'
+
+    def setup() {
```

Comment:
Didn't you mean `setupSpec`? Setup is run before every method.

Reply:
Yeah actually in these tests there is not need of setting up wiremock before each test.

Reply:
Then i have to make everything `@Shared` or static